system/GD32E23x_firmware/CMSIS/GD/GD32E23x/Source/system_gd32e23x.c

```diff
@@ -40,7 +40,7 @@
 #define __HXTAL           (HXTAL_VALUE)            /* high speed crystal oscillator frequency */
 #define __SYS_OSC_CLK     (__IRC8M)                /* main oscillator frequency */
 
-#define VECT_TAB_OFFSET  (uint32_t)0x00            /* vector table base offset */
+//#define VECT_TAB_OFFSET  (uint32_t)0x00            /* vector table base offset */
```

Comment:
I don't see a 

```
#ifndef VECT_TAB_OFFSET  
#define VECT_TAB_OFFSET ...
#endif 
```
here, would that not mean that this file tries to use a hardcoded vector table offset?

Reply:
It's pulled in from platform.txt. The variable `build.bootloader_flags=-DVECT_TAB_OFFSET={build.flash_offset}` wasn't being used anywhere. `flash_offset` is being used to inform the linker script through `LD_FLASH_OFFSET={build.flash_offset}`.

Reply:
But if `-DVECT_TAB_OFFSET={build.flash_offset}` is additionally set as global compiler settings, would that not cause a "VECT_TAB_OFFEST redefined" warning in GCC when compiling this file like in https://stackoverflow.com/a/73518725?

Reply:
Indeed it would, so the correct way for the user to override it would be setting build.flash_offset.